Tests/AppTests/BuildMonitorIndexModelTests.swift

```diff
@@ -0,0 +1,151 @@
+            let version = try Version(package: package,
+                                      latest: .defaultBranch,
+                                      packageName: "LeftPad from Version.packageName",
+                                      reference: .branch("main"))
+            try version.save(on: app.db).wait()
+            try Build(id: .id0,
+                      version: version,
+                      platform: .macosXcodebuild,
+                      runnerId: BuildRunner.mac0.rawValue,
+                      status: .ok,
+                      swiftVersion: .init(5, 6, 0)).save(on: app.db).wait()
+            try Repository(package: package,
+                           ownerName: "Dave Verwer from Repository.ownerName").save(on: app.db).wait()
+        }
+
+        // Query results back through the Joined4
+        let buildResult = try BuildResult.query(on: app.db).first().wait().unwrap()
+
+        // MUT
+        let model = try XCTUnwrap(BuildMonitorIndex.Model(buildResult: buildResult))
+
+        XCTAssertEqual(model.buildId, .id0)
+        XCTAssertEqual(model.packageName, "LeftPad from Version.packageName")
+        XCTAssertEqual(model.repositoryOwnerName, "Dave Verwer from Repository.ownerName")
+        XCTAssertEqual(model.platform, .macosXcodebuild)
+        XCTAssertEqual(model.swiftVersion, .init(5, 6, 0))
+        XCTAssertEqual(model.reference, .branch("main"))
+        XCTAssertEqual(model.referenceKind, .defaultBranch)
+        XCTAssertEqual(model.status, .ok)
+        XCTAssertEqual(model.runner, .mac0)
+    }
+
+    func test_init_from_Build_without_repository_name() throws {
+        do {
+            let package = try savePackage(on: app.db, "https://github.com/daveverwer/LeftPad")
+            let version = try Version(package: package
+                                      // Deliberately missing a `packageName`
+            )
+            try version.save(on: app.db).wait()
+            try Build(version: version,
+                      platform: .macosXcodebuild,
+                      status: .ok,
+                      swiftVersion: .init(5, 6, 0)).save(on: app.db).wait()
+            try Repository(package: package,
+                           name: "LeftPad from Repository.name").save(on: app.db).wait()
+        }
+
+        // Query results back through the Joined4
+        let buildResult = try BuildResult.query(on: app.db).first().wait().unwrap()
+
+        // MUT
+        let model = try XCTUnwrap(BuildMonitorIndex.Model(buildResult: buildResult))
+
+        XCTAssertEqual(model.packageName, "LeftPad from Repository.name")
+    }
+
+    func test_init_from_Build_with_no_package_name() throws {
+        do {
+            let package = try savePackage(on: app.db, "https://github.com/daveverwer/LeftPad")
+            let version = try Version(package: package
+                                      // Deliberately missing a `packageName`
```

Comment:
Perhaps self-document?
```suggestion
                                      packageName: nil
```

Reply:
We can add the `packageName: nil` in addition to the comment, if you like, but I think the comment is important. It wouldn't be clear that that line specifically is important over and above all the other setup.

Reply:
Well, I thought the test name `test_init_from_Build_without_repository_name` would help with that :)

Reply:
All parts of the puzzle for sure, but I know that if/when I ever need to come back to this test it'll help me understand it.

Reply:
I'll add the `packageName: nil` though